mirrormanager2/lib/model.py

```diff
@@ -827,25 +828,29 @@ def emergency_expire_old_file_details(self, session):
         CAUTION: this will make all slightly outdated mirrors be non-trusted,
         so should be used sparingly!
         """
+        if not self.directory:
+            return False
 
-        files = session.query(FileDetail). \
-            filter_by(directory_id=self.directory_id). \
-            order_by(FileDetail.timestamp.desc()).all()
+        subdirs = session.query(Directory). \
```

Comment:
I kinda like more the style where the braces are used to avoid using ``\``, something like:

````
session.query(
    Directory
).filter(
   ...
).all()
````
Same for below

But this is really just style